dbms/src/Debug/astToExecutor.cpp

```diff
@@ -1343,6 +1347,111 @@ void Join::toMPPSubPlan(size_t & executor_index, const DAGProperties & propertie
+{
+    tipb_executor->set_tp(tipb::ExecType::TypeWindow);
+    tipb_executor->set_executor_id(name);
+    tipb::Window * window = tipb_executor->mutable_window();
+    auto & input_schema = children[0]->output_schema;
+    for (const auto & expr : func_descs)
+    {
+        tipb::Expr * window_expr = window->add_func_desc();
+        const auto * window_func = typeid_cast<const ASTFunction *>(expr.get());
+        for (const auto & arg : window_func->arguments->children)
+        {
+            tipb::Expr * func = window_expr->add_children();
+            astToPB(input_schema, arg, func, collator_id, context);
+        }
+        window_expr->set_tp(tipb::ExprType::RowNumber);
+        auto * ft = window_expr->mutable_field_type();
+        ft->set_tp(TiDB::TypeLongLong);
+        ft->set_flag(TiDB::ColumnFlagBinary);
+        ft->set_collate(collator_id);
+        // rowNumber don't have children.
+        // ft->set_decimal(window_expr->children(0).field_type().decimal()); // ywq todo check type?
+        // ft->set_flen(window_expr->children(0).field_type().flen());
+    }
+    for (const auto & child : order_by_exprs)
+    {
+        auto * elem = typeid_cast<ASTOrderByElement *>(child.get());
+        if (!elem)
+            throw Exception("Invalid order by element", ErrorCodes::LOGICAL_ERROR);
+        tipb::ByItem * by = window->add_order_by();
+        by->set_desc(elem->direction < 0);
+        tipb::Expr * expr = by->mutable_expr();
+        astToPB(children[0]->output_schema, elem->children[0], expr, collator_id, context);
+    }
+
+    for (const auto & child : partition_by_exprs)
+    {
+        auto * elem = typeid_cast<ASTOrderByElement *>(child.get());
+        if (!elem)
+            throw Exception("Invalid order by element", ErrorCodes::LOGICAL_ERROR);
+        tipb::ByItem * by = window->add_partition_by();
+        by->set_desc(elem->direction < 0);
+        tipb::Expr * expr = by->mutable_expr();
+        astToPB(children[0]->output_schema, elem->children[0], expr, collator_id, context);
+    }
+
+    tipb::WindowFrame * mut_frame = window->mutable_frame();
+    if (frame.start.has_value())
+    {
+        auto * start = mut_frame->mutable_start();
+        start->set_offset(std::get<2>(frame.start.value()));
+        start->set_unbounded(std::get<1>(frame.start.value()));
+        start->set_type(std::get<0>(frame.start.value()));
+    }
+
+    if (frame.end.has_value())
+    {
+        auto * end = mut_frame->mutable_end();
+        end->set_offset(std::get<2>(frame.end.value()));
+        end->set_unbounded(std::get<1>(frame.end.value()));
+        end->set_type(std::get<0>(frame.end.value()));
+    }
+
+    mut_frame->set_type(frame.type);
+    auto * children_executor = window->mutable_child();
+    return children[0]->toTiPBExecutor(children_executor, collator_id, mpp_info, context);
+}
+
+void Window::columnPrune(std::unordered_set<String> & used_columns)
+{
```

Comment:
`window.schema != children[0]->schema`.
You can refer to `aggregate::columnPrune`